pkg/ui/react-app/src/pages/graph/PanelList.tsx

```diff
@@ -163,13 +168,16 @@ const PanelList: FC<RouteComponentProps & PathPrefixProps> = ({ pathPrefix = ''
           Error fetching metrics list: Unexpected response status when fetching metric names: {metricsErr.message}
         </Alert>
       )}
-      <PanelListContent
-        panels={decodePanelOptionsFromQueryString(window.location.search)}
-        pathPrefix={pathPrefix}
-        useLocalTime={useLocalTime}
-        metrics={metricsRes.data}
-        queryHistoryEnabled={enableQueryHistory}
-      />
+      {!storesLoading && (
```

Comment:
This will show a blank page until we fetch the details of the stores. You can use the `withStatusIndicator` function we have to show a status indicator until we get all the data. you can find many examples of its usage.